core/models/openimis_graphql_test_case.py

```diff
@@ -1,29 +1,92 @@
 import json
 from django.conf import settings
 from graphene_django.utils.testing import GraphQLTestCase
+from django.core.management import call_command
+from django.apps import apps
+from django.db import transaction, connection
+
+from django.test import TransactionTestCase
+from simple_history.models import HistoricalChanges
 import uuid
 import logging
 from graphene import Schema
 from graphene.test import Client
 import datetime
 import time
+from core.models import ObjectMutation, MutationLog
+
 
 logger = logging.getLogger(__name__)
 
+MODEL_NOT_TO_FLUSH = [
+    'gender',
+    'role',
+    'roleright',
+    'profession',
+    'familytype',
+    'location',
+    'healthfacility',
+    'permission'
+    
+]
+TABLE_TO_FLUSH = ['tblLogins', 'tblHealthStatus'] 
 
-class openIMISGraphQLTestCase(GraphQLTestCase):
+class openIMISGraphQLTestCase(TransactionTestCase):
     GRAPHQL_URL = f"/{settings.SITE_ROOT()}graphql"
     GRAPHQL_SCHEMA = True
+    
+    def query(self, query, **kwargs):
+        """
+        Execute a GraphQL query using the client.
+        """
+        return GraphQLTestCase.query.__call__(self, query, **kwargs)
 
     class BaseTestContext:
         def __init__(self, user):
             self.user = user
     # client = None
+    def setUp(self):
+        # cls.client=Client(cls.schema)
+        GraphQLTestCase.setUp.__call__(self)
+        super().setUp()
```

Comment:
The fact that this implementation is inheriting from one class and reaching out to another class that is meant to be inherited from with `__call__` multiple times, creates a code smell.

Reply:
that the only way I found in python to call an external with another context, inheritance don't work because  GraphQLTestCase inherite for TestCase, which inherite from TransactionTestCase ....  yes it smell